cmd/clusterctl/client/cluster/client.go

```diff
@@ -33,6 +33,9 @@ import (
 
 const (
 	minimumKubernetesVersion = "v1.16.0"
+	// firstUnsupportedKubernetesVersion is the first version we don't support as management cluster.
+	// We cannot support v1.22.0 as it's not compatible with our controller-runtime version and WebhookConfigurations v1beta1.
+	firstUnsupportedKubernetesVersion = "v1.22.0"
```

Comment:
It was easier to verify it that way. I guess i could also check serverVersion <= v1.21.99 (or something similar)